### DISCUSSION TITLE
Is Raspberry Pi Zero 2W supported?

### DISCUSSION THREAD

**Discussion by reubenmiller** [silitics/rugix#1] [Q&A] [answered]
I'm currently trying to flash the custom built image, however after flashing the SD card using the Raspberry Pi Imager, the Zero doesn't startup, the LED just blinks 4 times (which seems to indicate that the "start*.elf not found" - though not 100% sure as the docs don't explicitly mention the Zero 2W).

I tested the memory card with the vanilla Raspberry Pi OS image, and the same SD card works with a different image, so it seems to indicate that something is off with the build.

I built the image using the [rugpi-template](https://github.com/silitics/rugpi-template), and created my own project [](https://github.com/reubenmiller/rpi-tedge-image), which built with no issues.

Do you know if there is some issues with the Raspberry Pi Zero 2W, or some other EEPROM setting that needs to be done first to enable the tryboot functionality?

**Comment by reubenmiller** [comment]
Looking a bit more into this and it looks like rugpi is assuming that it is a RPI4 which uses the rpi-eeprom, where as Raspberry PI Zeros 2W whilst being an arm64 architecture, uses the [bootcode.bin](https://www.raspberrypi.com/documentation/computers/raspberry-pi.html#raspberry-pi-4-boot-eeprom)

**Comment by koehlma** [comment]
No, at the moment, only Raspberry Pi 4 and the Compute Module 4 are supported. I think that the `tryboot` feature could be replicated on the other boards using U-Boot, however, I did not yet look further into this. I am going to update the documentation to clarify that currently only Raspberry Pi 4 is supported.

**Reply by reubenmiller** [reply]
Thanks for the confirmation. Though generally I really like the build flow, it much easier to get a custom build quickly than yocto.

I'm just investigating reliable OS (firmware) update mechanisms as I'm the the product owner of [thin-edge.io](https://github.com/thin-edge/thin-edge.io) and we're wanting to add cloud support for triggering these updates in the near future.

**Reply by koehlma** [reply]
Thanks! I am glad you like the build flow. Obviously, this is still an early prototype, but we would love to add support for other boards in the future (including other boards than Raspberry Pi), if there is sufficient demand. :)

Thin-edge sounds interesting. I will definitely keep an eye on it. We actually have a Mender integration for Rugpi to trigger/deploy updates via Mender. However, this integration is not open-source yet.

**Comment by koehlma** [comment, marked as answer]
Rugpi now supports Pi Zero 2W through U-Boot integration.
